core/src/main/java/com/datastrato/gravitino/authorization/AuthorizationManager.java

```diff
@@ -0,0 +1,261 @@
+                  LOG.info("Closing authorization {}.", k);
+                  ((AuthorizationWrapper) v).close();
+                })
+            .scheduler(
+                Scheduler.forScheduledExecutorService(
+                    new ScheduledThreadPoolExecutor(
+                        1,
+                        new ThreadFactoryBuilder()
+                            .setDaemon(true)
+                            .setNameFormat("authorization-cleaner-%d")
+                            .build())))
+            .build();
+  }
+
+  @Override
+  public void close() {
+    authorizationCache.invalidateAll();
+  }
+
+  public AuthorizationWrapper loadAuthorizationAndWrap(CatalogEntity entity)
+      throws NoSuchAuthorizationException {
+    return authorizationCache.get(entity, this::createAuthorizationWrapper);
+  }
+
+  /** Wrapper class for an Authorization instance and its class loader. */
+  public static class AuthorizationWrapper {
+    private BaseAuthorization authorization;
+
+    private IsolatedClassLoader classLoader;
+
+    public AuthorizationWrapper(BaseAuthorization auth, IsolatedClassLoader classLoader) {
+      this.authorization = auth;
+      this.classLoader = classLoader;
+    }
+
+    public <R> boolean runAuthorizationChain(Function<AuthorizationOperations, R>... functions) {
+      try {
+        return classLoader.withClassLoader(
+            cl -> {
+              for (Function<AuthorizationOperations, R> function : functions) {
+                function.apply(authorization.ops());
+              }
+              return true;
+            });
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    @VisibleForTesting
+    public AuthorizationOperations getOps() {
+      return authorization.ops();
+    }
+
+    public void close() {
+      try {
+        classLoader.withClassLoader(
+            cl -> {
+              if (authorization != null) {
+                authorization.close();
+              }
+              authorization = null;
+              return null;
+            });
+      } catch (Exception e) {
+        LOG.warn("Failed to close authorization", e);
+      }
+
+      classLoader.close();
+    }
+  }
+
+  @SafeVarargs
+  public final <R> boolean runAuthorizationChain(
+      CatalogEntity entity, Function<AuthorizationOperations, R>... functions) {
+    return loadAuthorizationAndWrap(entity).runAuthorizationChain(functions);
+  }
+
+  private AuthorizationWrapper createAuthorizationWrapper(CatalogEntity entity) {
+    Map<String, String> conf = entity.getProperties();
+    String provider = conf.get(AUTHORIZATION_PROVIDER);
+
+    IsolatedClassLoader classLoader = createClassLoader(provider, conf);
+    BaseAuthorization<?> baseAuthorization = createBaseAuthorization(classLoader, entity);
+
+    return new AuthorizationWrapper(baseAuthorization, classLoader);
+  }
+
+  private IsolatedClassLoader createClassLoader(String provider, Map<String, String> conf) {
```

Comment:
This logic seems to be similar to `CatalogManager`, can we reuse some code

Reply:
Yes, I moved some common functions into `IsolatedClassLoader class`.